src/test/java/it/gov/pagopa/reward/notification/service/rewards/evaluate/notify/RewardsNotificationExpiredInitiativeHandlerServiceIntegrationTest.java

```diff
@@ -0,0 +1,117 @@
+import java.util.List;
+
+
+class RewardsNotificationExpiredInitiativeHandlerServiceIntegrationTest extends BaseIntegrationTest {
+
+    public static final LocalDate TODAY = LocalDate.now();
+
+    private RewardNotificationRule expiredInitiative;
+    private RewardNotificationRule validInitiative;
+
+    private RewardsNotification expiredNotification1;
+    private RewardsNotification expiredNotification2;
+    private RewardsNotification validNotification1;
+    private RewardsNotification validNotification2;
+
+    @Autowired
+    private RewardNotificationRuleRepository ruleRepository;
+    @Autowired
+    private RewardsNotificationRepository notificationRepository;
+    @Value("${app.rewards-notification.expired-initiatives.day-before}")
+    private int dayBefore;
+
+    private RewardsNotificationExpiredInitiativeHandlerService service;
+
+    private void storeTestData() {
+        // initiatives
+        expiredInitiative = ruleRepository.save(
+                RewardNotificationRuleFaker.mockInstanceBuilder(1)
+                        .initiativeId("INITIATIVE1")
+                        .endDate(TODAY.minusDays(5))
+                        .build()
+        ).block();
+        validInitiative = ruleRepository.save(
+                RewardNotificationRuleFaker.mockInstanceBuilder(2)
+                        .initiativeId("INITIATIVE2")
+                        .endDate(TODAY.plusDays(30))
+                        .build()
+        ).block();
+
+        // notifications
+        expiredNotification1 = notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(1)
+                        .initiativeId("INITIATIVE1")
+                        .notificationDate(TODAY.minusDays(6))
+                        .build()
+        ).block();
+        expiredNotification2 = notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(2)
+                        .initiativeId("INITIATIVE1")
+                        .notificationDate(null)
+                        .build()
+        ).block();
+        validNotification1 = notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(3)
+                        .initiativeId("INITIATIVE2")
+                        .notificationDate(TODAY)
+                        .build()
+        ).block();
+        validNotification2 = notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(4)
+                        .initiativeId("INITIATIVE2")
+                        .notificationDate(null)
+                        .build()
+        ).block();
+    }
+
+    @BeforeEach
+    void init() {
+        service = new RewardsNotificationExpiredInitiativeHandlerServiceImpl(ruleRepository, notificationRepository, dayBefore);
+        storeTestData();
+    }
+
+    @AfterEach
+    void clearData() {
+        ruleRepository.deleteAll().block();
+        notificationRepository.deleteAll().block();
```

Comment:
store the saved entities in lists, and here delete just that records.
This way we could create a TestIntegrated version of this test, thus testing it into cosmosdb